components/service/glean/src/test/java/mozilla/components/service/glean/debug/GleanDebugActivityTest.kt

```diff
@@ -140,4 +140,102 @@ class GleanDebugActivityTest {
 
         server.shutdown()
     }
+
+    @Test
+    fun `pings are tagged using tagPings`() {
+        val server = MockWebServer()
+        server.enqueue(MockResponse().setBody("OK"))
+
+        resetGlean(config = Glean.configuration.copy(
+            serverEndpoint = "http://" + server.hostName + ":" + server.port
+        ))
+
+        // Put some metric data in the store, otherwise we won't get a ping out
+        // Define a 'booleanMetric' boolean metric, which will be stored in "store1"
+        val booleanMetric = BooleanMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Application,
+            name = "boolean_metric",
+            sendInPings = listOf("store1")
+        )
+
+        booleanMetric.set(true)
+        assertTrue(booleanMetric.testHasValue())
+
+        // Set the extra values and start the intent.
+        val intent = Intent(ApplicationProvider.getApplicationContext<Context>(),
+            GleanDebugActivity::class.java)
+        intent.putExtra(GleanDebugActivity.SEND_PING_EXTRA_KEY, "store1")
+        intent.putExtra(GleanDebugActivity.TAG_DEBUG_VIEW_EXTRA_KEY, "test-debug-ID")
+        val activity = Robolectric.buildActivity(GleanDebugActivity::class.java, intent)
+        activity.create().start().resume()
+
+        // Since we tagged the ping it will override the serverEndpoint to the GCP endpoint so we
+        // need to re-override it here.
+        Glean.configuration = Glean.configuration.copy(
+            serverEndpoint = "http://" + server.hostName + ":" + server.port
+        )
+
+        triggerWorkManager()
+        val request = server.takeRequest(10L, TimeUnit.SECONDS)
+
+        assertTrue(
+            "Request path must be correct",
+            request.requestUrl.encodedPath().startsWith("/submit/mozilla-components-service-glean/store1")
+        )
+
+        assertEquals(
+            "Headers must contain X-Debug-ID",
+            "test-debug-ID",
+            request.headers.get("X-Debug-ID")
+        )
+
+        server.shutdown()
+    }
+
+    @Test
+    fun `tagPings filters ID's that don't match the pattern`() {
```

Comment:
Let's also check in here (or a separate test) that the URL is correctly reset to the previous value in this case.

Reply:
I assert that it already does this.   Since the config is set to the local server endpoint at the beginning of this test, and it doesn't need to be reset in order to capture the ping to see that it doesn't contain the header, this indeed tests that the server endpoint matches the configuration endpoint.  Otherwise the ping would never be received.

Reply:
Would that cover the case highlighted [here](https://github.com/mozilla-mobile/android-components/pull/2441#discussion_r268679816)?

Reply:
Added coverage to verify the endpoint is reset.  Because of how I reset it I had to touch a couple of other tests as well to keep them working.